java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -131,6 +171,58 @@ public Block(int id) {
     public Tree terminator() {
       return terminator;
     }
+
+    public boolean notActive() {
+      return terminator == null && elements.isEmpty();
+    }
+
+    private void prune(Block inactiveBlock) {
+      if (inactiveBlock == trueBlock) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1698)